### ISSUE TITLE
LXD should be able to create its own images

### ISSUE THREAD

**Issue by tych0** [lxc/lxd#311]
A feature request from a user: it would be handy if LXD could create its own images, i.e. you just upload a tarball and post some extra metadata (e.g. as a multipart post or something), and LXD sews it all together for you and generate the image, and tells you what the hash, etc. is.

**Comment by stgraber** [comment]
The spec currently defines how to turn a container into an image by feeding a json dict and how to import an image from a lxd image tarball. The combination of both using multipart post + some repacking code would implement this.

Seems like a nice to have and not urgent, so targeting at 0.5 for now, may be pushed back even further.

**Comment by mynameisdaniil** [comment]
Any progress there? I can't even create image from container/snapshot using REST API.
```
curl -XPOST localost:8080/1.0/images -d@container.json
```
returns 500/Unsupported compression.
Seems like there is no way to create my own base images.

**Comment by rcarmo** [comment]
I'd like to be able to create my own base images as well (Ubuntu armhf here).

**Comment by pcdummy** [comment]
In current master theres a way to create your own images:

    lxc publish <server>:<container> <server>: --alias ubuntu/trusty -- alias ubuntu/trusty/amd64

So you create a container from a given base image, modify the container and publish it back to the image registry, nice isn't it?

I've a [blog post](http://rene.jochums.at/2015/07/07/lxd-getting-started/) about it.

**Comment by stgraber** [comment]
Right, that's one part of it. The other part is the branch I'm working on now which implements split images where you have one tarball for the metadata and one for the rootfs. Once that lands, we can close that issue for good.
